integration-tests/management-interface/src/main/resources/application.properties

```diff
@@ -0,0 +1 @@
+quarkus.management.enabled=true
```

Comment:
@cescoffier Can you please have this test confirming the optional authentication works ?

Reply:
@cescoffier I noticed `AuthConfig` was removed which was fine, but IMHO we really should have something available from the start, having an open management port in prod mode without any simple authentication is a problem. Complexity of the MTLS setup to have a secure management port access would indeed be rarely used if ever used (esp if we assume the browser will be used to access it).
We need to have a Basic Auth support from the start IMHO. You said Verx authentication would be used directly, not sure it is a good idea as it would prevent reusing Quarkus identity providers like JDBC/JPA to keep user name and passwords. It should not really bypass the actual Quarkus Security.

I can propose a fairly simple solution:

* Have `ManagementAuthConfig` group, but only copy `AuthConfig.basic` to it, so that if someone wants they just do `quarkus.management.auth.basic=true`
* If it is enabled then BasicAuthenticationMechanism is created (update the condition in HttpSecurityRecorder where it is created) and then in HttpSecurityRecorder.initPermissions, have `PathMatchingHttpSecurityPolicy` updated with one more `PolicyMappingConfig` (add new method to PathMatchingHttpSecurityPolicy to accept `PolicyMappingConfig` which will have `basic` authentication scheme requiring an authenticates access only).

And that should be it for now, then users just will have to add security-jpa or security-jdbc.

Would you like to give it a quick try, it will literally take 30 mins for you to complete.

In the future we can also update those policies to be applicable to specific ports only, I can handle the port related part separately.

Reply:
Also if for ex both management and application need Basic auth, then how to ensure that someone who passes the application level basic auth check can not access management, and vice versa, for that PolicyMappingConfig for management can in the future include a `management` role, it can be taken care of in the future

Reply:
I agree that basic auth would be nice, but I don't have the time to do it right now. 
Also, I've looked into hooking the quarkus security provider, and it requires a lot more work and adaptation than just using vertx auth.

Reply:
@sberyozkin it would have the change somewhere I'm happy to have a look. My initial investigation really scared me.

Reply:
@cescoffier Sure :-), will try to have a look asap, if it gets too messy then we can indeed follow up without blocking this PR, thanks

Reply:
Please resolve the comment to minimize the PR noise if you'd like to, thanks

Reply:
@cescoffier I've been working on fixing the test today, but haven't achieved much, may be because I'm trying to do too many things today. I'm travelling after lunch tomorrow so will try to focus in the morning on it. As I said, if it is the only thing that holds this PR then please lift the change request and I'll follow with the PR when ready. Thanks

Reply:
@cescoffier Can you clarify something please, what happens in `HttpSecurityRecorder`, does it affect the main router only or other routers such as the management router is also affected ?

Reply:
Only the main router.

Reply:
@cescoffier Thanks, I started guessing it was the case, after getting the basic authentication activated only in the 404 case  in the test which I understand now hits the main router only. 
What exactly binds to the main router to it, I don't see any obvious link, 

I see in `HttpSecurityProcessor`:

```
@BuildStep
 @Record(ExecutionTime.RUNTIME_INIT)
    SyntheticBeanBuildItem initBasicAuth(
            HttpSecurityRecorder recorder,
            HttpBuildTimeConfig buildTimeConfig...) {
...
}
```
and
```
public HttpSecurityRecorder(RuntimeValue<HttpConfiguration> httpConfiguration, HttpBuildTimeConfig buildTimeConfig) {
        this.httpConfiguration = httpConfiguration;
        this.buildTimeConfig = buildTimeConfig;
 }
```

Both routers are created in `VertxHttpRecorder` but I lack some Vert.x HTTP extension knowledge to understand why `HttpSecurityProcessor` and `HttpSecurityRecorder` affect the main router only. 
Can you clarify please ? It is obviously taking more more than my 30 mins's assertion, so I'll lift my change request. I'll continue investigating the Quarkus Sec path anyway, but I'd appreciate some help related to this router binding, thanks

Reply:
@sberyozkin You would need to completely duplicate the `HttpSecurityRecorder` because the input of that class are the main server configuration and not the management one. However, the `authenticationMechanismHandler` should be reusable (at 90%).

However, with a bit of modification, I believe we would be able to reduce the amount of duplication to almost nothing:

1 - modify the current recorder to extract the used value from the received configuration
2 - add a second recorded for the management interface doing the same
3 - extract the field containing these values and the `authenticationMechanismHandler` in a parent class.
4- call the new recorder when we configure the management interface

It's important to understand that the main and management servers are managed entirely separately at runtime, on purpose.

Reply:
@cescoffier Thanks, I'll try to get something working with a duplicate first just to get some feeling it actually works, and then we can optimize, will ping offline for some extra hints, but in any case, as I said, I don't want to hold this PR, my change request has been lifted